superset/views/core.py

```diff
@@ -1410,10 +1404,7 @@ def filter(self, datasource_type, datasource_id, column):
             datasource_type, datasource_id, db.session)
         if not datasource:
             return json_error_response(DATASOURCE_MISSING_ERR)
-        if not security_manager.datasource_access(datasource):
-            return json_error_response(
-                security_manager.get_datasource_access_error_msg(datasource))
-
+        self.check_datasource_permission(datasource)
```

Comment:
Same as mentioned above, this looks like it assumes that check_datasource_permission raises

Reply:
ack